packages/opentelemetry-instrumentation-grpc/src/instrumentation.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+import { GrpcInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import { GrpcNativeInstrumentation } from './grpc';
+import { GrpcJsInstrumentation } from './grpc-js';
+import * as api from '@opentelemetry/api';
+import { MeterProvider } from '@opentelemetry/api-metrics';
+
+/** The metadata key under which span context is stored as a binary value. */
+export const GRPC_TRACE_KEY = 'grpc-trace-bin';
+
+export class GrpcInstrumentation {
+  private _grpcNativeInstrumentation: GrpcNativeInstrumentation;
+  private _grpcJsInstrumentation: GrpcJsInstrumentation;
+
+  public readonly instrumentationName: string =
+    '@opentelemetry/instrumentation-grpc';
+  public readonly instrumentationVersion: string = VERSION;
+
+  constructor(
+    protected _config: GrpcInstrumentationConfig & InstrumentationConfig = {}
+  ) {
+    this._grpcJsInstrumentation = new GrpcJsInstrumentation(
+      _config,
+      this.instrumentationName,
+      this.instrumentationVersion
+    );
+    this._grpcNativeInstrumentation = new GrpcNativeInstrumentation(
+      _config,
+      this.instrumentationName,
+      this.instrumentationVersion
+    );
+  }
+
+  public setConfig(
+    config: GrpcInstrumentationConfig & InstrumentationConfig = {}
+  ) {
+    this._config = Object.assign({}, config);
+    this._grpcJsInstrumentation.setConfig(this._config);
+    this._grpcNativeInstrumentation.setConfig(this._config);
+  }
+
+  /**
+   * @internal
+   * Public reference to the protected BaseInstrumentation `_config` instance to be used by this
+   * plugin's external helper functions
+   */
+  public getConfig() {
+    return this._config;
+  }
+
+  init() {
+    // sub instrumentations will already be init when constructing them
+    return;
+  }
+
+  enable() {
+    this._grpcJsInstrumentation.enable();
+    this._grpcNativeInstrumentation.enable();
+  }
+
+  disable() {
+    this._grpcJsInstrumentation.disable();
+    this._grpcNativeInstrumentation.disable();
+  }
+
+  /**
+   * Sets MeterProvider to this plugin
+   * @param meterProvider
+   */
+  public setMeterProvider(meterProvider: MeterProvider) {
```

Comment:
Seems this is the only reason for the adding `@opentelemetry/api-metrics` to dependencies.

Do we really need this? Maybe remove this API or if it is needed to fulfill the interface it could be a dummy.

Reply:
It could be used if at some point we want to add metrics into instrumentations, thats one of the reason why we re-implemented the plugin system to have what we have here.

Reply:
Sure, we could use it in future. But why add the dependency already now and not once we need it?

Reply:
It's already a dependency for `@opentelemetry/api-metrics`, the goal for the plugin is to export the exact same interface as `BaseInstrumentation`, i'm just re-importing explicity to avoid issues.